Microsoft.AVS.VVOLS/Microsoft.AVS.VVOLS.psm1

```diff
@@ -0,0 +1,370 @@
+        [PSCredential]
+        $ProviderCredential
+    )
+
+    $VasaProvider = Get-VasaProvider -Name $ProviderName -ErrorAction Ignore
+    if ($vasaProvider) {
+        throw "Unable to create a VASA provider. Vasa provider '$ProviderName' already exists."
+    }
+
+    New-VasaProvider -Name $ProviderName -Credential $ProviderCredential -Url $ProviderUrl -Force -ErrorAction Stop | Out-Null
+
+    Write-Host "VASA provider $ProviderName created successfully."
+}
+
+<#
+    .SYNOPSIS
+     Remove a VASA provider from vCenter
+
+    .PARAMETER ProviderName
+     Name of VASA provider to be removed from vCenter
+
+    .EXAMPLE
+     Remove-VvolVasaProvider -ProviderName "myProvider"
+
+    .INPUTS
+     vCenter VASA provider name
+
+    .OUTPUTS
+     None.
+#>
+function Remove-VvolVasaProvider {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Name of VASA provider to be created in vCenter')]
+        [ValidateNotNull()]
+        [String]
+        $ProviderName
+    )
+
+    $VasaProvider = Get-VasaProvider -Name $ProviderName -ErrorAction Ignore
+    if (-not $VasaProvider) {
+        throw "Unable to remove a VASA provider. Vasa provider '$ProviderName' does not exist."
+    }
+
+    Remove-VasaProvider -Provider $VasaProvider -Confirm:$false -ErrorAction Stop | Out-Null
+
+    Write-Host "VASA provider $ProviderName removed successfully."
+}
+
+
+<#
+    .SYNOPSIS
+     Creates a new vVol Storage Policy
+
+    .PARAMETER PolicyConfigJsonString
+     Storage policy in Json string. The command will traverse through the SPBM rules and creat a storage policy containing all of the rules.
+     The example string is as below:
+    {
+        "SchemaVersion": "1.0.0",
+        "Vendor": "Pure Storage",
+        "PolicyName": "",
+        "PolicyDescription": "",
+        "SpbmRules": {
+            "com.purestorage.storage.policy.FlashArrayGroup":["fa1","fa2"],
+            "com.purestorage.storage.replication.LocalSnapshotPolicyCapable": true,
+            "com.purestorage.storage.replication.LocalSnapshotInterval":"00:00:00"
+            ...
+        }
+    }
+
+    .EXAMPLE
+     New-VvolStoragePolicy -PolicyConfigJsonString $policyConfigJsonString
+
+    .INPUTS
+     vCenter storage policy in Json string
+
+    .OUTPUTS
+     None.
+#>
+function New-VvolStoragePolicy {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param(
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Storage policy in Json string')]
+        [ValidateNotNull()]
+        [String]$PolicyConfigJsonString
+    )
+    # Convert JSON to PowerShell object
+    $PolicyConfig = ConvertFrom-Json $PolicyConfigJsonString
+    # Extract values from the object
+    $Vendor = $PolicyConfig.Vendor
+    $PolicyName = $PolicyConfig.PolicyName
+    $PolicyDescription = $PolicyConfig.PolicyDescription
```

Comment:
Could we have a check at the beginning of this cmdlet to ensure that $Vendor, $PolicyName and $PolicyDescription values are not empty?

Reply:
Added check for PolicyName
Vendor and description are optional